cpu/atmega_common/Makefile.features

```diff
@@ -1,6 +1,8 @@
+FEATURES_PROVIDED += adc_ng
 FEATURES_PROVIDED += arch_8bit
 FEATURES_PROVIDED += arch_avr8
 FEATURES_PROVIDED += atmega_pcint0
+FEATURES_PROVIDED += periph_adc_ng
```

Comment:
Should be defined in board definitions. Otherwise, a board is not able to use only an external ADC if the feature is provided by the MCU because of the following condition in `adc_ng.c`: https://github.com/RIOT-OS/RIOT/blob/4661b927af876f4beb61d4d074d33ef2b7b5acf8/drivers/adc_ng/adc_ng.c#L24

Reply:
Features and modules are conceptional not the same, even though for most (all?) features with `periph_%` a module with the same name exists and is automatically used, if the corresponding feature is used.

So: Just because the feature is provided, it is not automatically used. The `FEATURES_OPTIONAL += periph_adc_ng` in `drivers/Makefile.dep` that is done when `adc_ng` is used would indeed pull that feature (and therefore) the corresponding module in. But that could be prevented by `FEATURES_BLACKLIST += periph_adc_ng`.

I'm not sure if this is super elegant. Maybe it would be better to provide an `adc_ng_default` pseudo-module that would pull in all ADC-NG backends supported by a board.